Docker compose: Run Jupyter by default and expose port

AdamGleave opened the pull request:
Quality of life improvement to `docker-compose.yml` to start Jupyter notebook automatically (no need to `docker exec` and run it manually), and expose the port (no need to mess around with extra port forwarding or ngrok; just need to do SSH forwarding after this).

UFO-101 commented on notebooks/docker-compose.yml:
```diff
@@ -21,3 +21,9 @@ services:
+    ports:
+      # Host will chose a random available port. To find the port allocated, run
+      # docker-compose -f notebooks/docker-compose.yml port notebooks 8888
+      - 127.0.0.1::8888
```
Do we need the first bit? It took me a while to understand what was going on.
This would be clearer to me:
```
ports:
  - "8888"
```
Also to find port the quickest way is `docker ps` right? It auto filters to your own containers.

AdamGleave replied:
If you don't include the `127.0.0.1` then it will expose the port on public interfaces, which seems like an insecure default to have. I can add a comment explaining what it does.

UFO-101 replied:
Oh okay thanks, good to know.

UFO-101 commented:
Btw I guess you don't use VSCode cause we already have a devcontainer setup for that. What editor do you use?

tomtseng commented:
We should update the `notebooks/README.md` — I wrote some stuff for how to launch the notebooks without VSCode there,  but I didn't realize that there was already this docker-compose file for launching the notebooks more conveniently

AdamGleave commented:
Thanks for the review, team.

@tomtseng I've updated the README now, thanks for catching that.

@UFO-101 I've updated comment to explain why binding to 127.0.0.1, and to use `docker ps` instead. I usually use PyCharm, but was just connecting to the Jupyter notebooks directly. Nothing against VSCode, but seems desirable to have an editor-agnostic workflow.
